File: src/main/java/com/jforex/kforexutils/jfclient/KForexUtilsStrategy.java
package com.jforex.kforexutils.jfclient;

import com.dukascopy.api.*;
import com.jforex.kforexutils.misc.KForexUtils;

@RequiresFullAccess
@Library("D:/data/Projects/repos/KForexTestRunner/build/libs/KForexTestRunner-1.0-SNAPSHOT-all.jar")
public class KForexUtilsStrategy implements IStrategy {
    private IEngine engine;
    private IConsole console;
    private IHistory history;
    private IContext context;
    private IUserInterface userInterface;
    private KForexUtils kForexUtils;
    private JFClientBridge bridge;

    public void onStart(IContext context) {
        this.engine = context.getEngine();
        this.console = context.getConsole();
        this.history = context.getHistory();
        this.context = context;
        this.userInterface = context.getUserInterface();
        this.kForexUtils = new KForexUtils(context);
        this.bridge = new MyJFClientExampleClass();

        bridge.onStart(kForexUtils);
    }

    public void onAccount(IAccount account) {
    }

    public void onMessage(IMessage message) {
        kForexUtils.getMessageGateway().onMessage(message);
    }

    public void onStop() {
        bridge.onStop();
    }

    public void onTick(Instrument instrument,
                       ITick tick) {
    }

    public void onBar(Instrument instrument,
                      Period period,
                      IBar askBar,
                      IBar bidBar) {
    }
}

File: src/main/kotlin/com/jforex/kforexutils/strategy/KForexUtilsStrategy.kt
package com.jforex.kforexutils.strategy

import com.dukascopy.api.*
import com.jforex.kforexutils.misc.initKForexUtils
import com.jforex.kforexutils.misc.kForexUtils
import com.jforex.kforexutils.price.BarQuote
import com.jforex.kforexutils.price.TickQuote


class KForexUtilsStrategy : IStrategy
{
    override fun onStart(context: IContext)
    {
        initKForexUtils(context)
    }

    override fun onMessage(message: IMessage)
    {
        kForexUtils
            .messageGateway
            .onMessage(message)
    }

    override fun onBar(
        instrument: Instrument,
        period: Period,
        askBar: IBar,
        bidBar: IBar
    )
    {
        val barQuote = BarQuote(
            instrument,
            period,
            askBar,
            bidBar
        )
        kForexUtils.onBarQuote(barQuote)
    }

    override fun onTick(
        instrument: Instrument,
        tick: ITick
    )
    {
        val tickQuote = TickQuote(instrument, tick)
        kForexUtils.onTickQuote(tickQuote)
    }

    override fun onStop()
    {
    }

    override fun onAccount(account: IAccount?)
    {
    }
}